languages/swift/exercises/concept/enums/.docs/introduction.md

```diff
@@ -0,0 +1,126 @@
+Enums in Swift are a mechanism of creating new types which are inhabited by a finite number of named values which may carry additional associated information, and can have properties and methods attached to them.
+
+### Defining Enums
+
+The most basic enums are defined as seen below.
+
+```swift
+enum NESButton {
+  case up
+  case down
+  case left
+  case right
+  case a
+  case b
+  case select
+  case start
+}
+```
+
+For brevity, multiple cases can also be written on a single line, separated by commas, with a single `case` keyword.
+
+```swift
+enum NESButton {
+  case up, down, left, right, a, b, select, start
+}
+```
+
+This defines a new type named `NESButtons` with possible values `up`, `down`, `left`, `right`, `a`, `b`, `select`, and `start`. These values can be referred to by following the name of the type followed by a dot (`.`) and the value. In cases where the type name can be inferred, only the dot and value are needed.
+
+### Methods
+
+Like other types in Swift, enums may contain methods which allow the enum to provide functionality based on the current value of the enum.
+
+Methods are analogous to functions, only they are defined inside the body of the enum and they are tied to the current enum value. They are accessed via _dot notation_ where the name of the enum value is followed by a dot (`.`) and the name of the method and its parameters.
+
+Inside the method, the enum value can be referred to as `self`, and in the type signature, if one is accepting as a parameter or returning a value of the the enum they can refer to the type as `Self`.
```

Comment:
It feels like this (`methods`) is a concept that is being unlocked here. Maybe add it to the `concepts` that are unlocked?

Reply:
I'm not sure of the best way to handle this. Both `enums` and `structs-and-classes` introduce methods. They are both exercises that have the same set of prerequisites, and one doesn't really come before the other (and it's too big of an exercise if they are combined). There is also a planned `methods` exercise that has both `enums` and `structs-and-classes` as prerequisites that will go into more detail on methods as well as introducing things like type (static) methods.

My thoughts were that it would be better to put the the `methods` concept (really `instance-methods` and `type-methods` concepts) in the `methods` exercise where they are more fully covered.

Reply:
> They are both exercises that have the same set of prerequisites, and one doesn't really come before the other (and it's too big of an exercise if they are combined).

So what I think I would do is just pick one of `enums` and `structs-and-classes` as the exercise to explain methods, and then have the other exercise depend on it (and remove the explanation of methods from the introduction/after). To me I would find it slightly confusing to have the same concept explained twice. 

> My thoughts were that it would be better to put the the methods concept (really `instance-methods` and `type-methods` concepts) in the methods exercise where they are more fully covered.

You've hit the nail on the head with this comment. I would suggest (see above) that either `enums` _or_ `structs-and-classes` unlocks the `methods` concept and then have the later `methods` exercise unlock the `instance-methods` and `type-methods` concepts.